controllers/helpers.go

```diff
@@ -149,7 +150,6 @@ func findVMByName(client *nutanixClientV3.Client, vmName string) (*nutanixClient
 }
 
 func getPEUUID(client *nutanixClientV3.Client, peName, peUUID *string) (string, error) {
```

Comment:
note: the pointers might not be nil by the value could be a zero value so consider checking for both.